src/lib/util/opresolv.cpp

```diff
@@ -19,546 +19,465 @@
 {
-	const char *option_specification;
-	error err;
-
-	for (auto &entry : m_entries)
-	{
-		if (entry.state == entry_state::UNSPECIFIED)
-		{
-			switch(entry.guide_entry->option_type) {
-			case OPTIONTYPE_INT:
-			case OPTIONTYPE_ENUM_BEGIN:
-				option_specification = lookup_in_specification(m_specification, entry.guide_entry);
-				assert(option_specification);
-				entry.set_int_value(-1);
-				err = resolve_single_param(option_specification, &entry, nullptr, 0);
-				if (err != error::SUCCESS)
-					return err;
-				break;
-
-			case OPTIONTYPE_STRING:
-				entry.value = "";
-				break;
-
-			default:
-				assert(FALSE);
-				return error::INTERNAL;
-			}
-			entry.state = entry_state::SPECIFIED;
-		}
-	}
-	return error::SUCCESS;
+	// NYI
+	return error::INTERNAL;
 }
 
 
 // -------------------------------------------------
-//  lookup_entry
+//	error_string
 // -------------------------------------------------
 
-const option_resolution::entry *option_resolution::lookup_entry(int option_char) const
+const char *option_resolution::error_string(option_resolution::error err)
 {
-	for (auto &entry : m_entries)
+	switch (err)
 	{
-		switch(entry.guide_entry->option_type) {
-		case OPTIONTYPE_INT:
-		case OPTIONTYPE_STRING:
-		case OPTIONTYPE_ENUM_BEGIN:
-			if (entry.guide_entry->parameter == option_char)
-				return &entry;
-			break;
-
-		default:
-			assert(FALSE);
-			return nullptr;
-		}
+	case error::SUCCESS:				return "The operation completed successfully";
+	case error::OUTOFMEMORY:			return "Out of memory";
+	case error::PARAMOUTOFRANGE:		return "Parameter out of range";
+	case error::PARAMNOTSPECIFIED:		return "Parameter not specified";
+	case error::PARAMNOTFOUND:			return "Unknown parameter";
+	case error::PARAMALREADYSPECIFIED:	return "Parameter specified multiple times";
+	case error::BADPARAM:				return "Invalid parameter";
+	case error::SYNTAX:					return "Syntax error";
+	case error::INTERNAL:				return "Internal error";
 	}
 	return nullptr;
 }
 
 
 // -------------------------------------------------
-//  lookup_int
+//	entry::ctor
 // -------------------------------------------------
 
-int option_resolution::lookup_int(int option_char) const
+option_resolution::entry::entry(const option_guide::entry &guide_entry)
+	: m_guide_entry(guide_entry), m_is_pertinent(false)
 {
-	auto entry = lookup_entry(option_char);
-	return entry ? entry->int_value() : -1;
 }
 
 
 // -------------------------------------------------
-//  lookup_string
+//	entry::set_enum_value_range
 // -------------------------------------------------
 
-const char *option_resolution::lookup_string(int option_char) const
+void option_resolution::entry::set_enum_value_range(option_guide::entrylist::const_iterator begin, option_guide::entrylist::const_iterator end)
 {
-	auto entry = lookup_entry(option_char);
-	return entry ? entry->value.c_str() : nullptr;
+	m_enum_value_begin = begin;
+	m_enum_value_end = end;
```

Comment:
The idea of assuming these iterators will remain valid for the lifetime of this object is really scary.  It just doesn't seem right to write code like this.